src/App.js

```diff
@@ -0,0 +1,178 @@
+import React, { useEffect, useState } from 'react';
+import InputRange from 'react-input-range';
+import 'react-input-range/lib/css/index.css'
+import logo from './jurgisLogo.png';
+import bgimage from './dp.jpg'
+import MovieCard from './components/MovieCard.js';
+import Spinner from 'react-bootstrap/Spinner'
+import 'bootstrap/dist/css/bootstrap.min.css';
+import './App.css';
+import { Jumbotron, Button, Navbar, Nav, NavDropdown, FormControl, } from "react-bootstrap";
+import ReactModal from 'react-modal';
+import YouTube from '@u-wave/react-youtube';
+
+let apikey = process.env.REACT_APP_JURGIS;
+let keyword = '';
+let page = 1;
+let movieList = [];
+
+function App() {
+  let [movies, setMovies] = useState(null);
+  const [genres, setGenres] = useState([]);
+  let [modal, SetModal] = useState(false);
+  let [rate, setRate] = useState(0);
+  let [trailer, setTrailer] = useState('');
+
+  let popularMovies = async () => {
+    let url = `https://api.themoviedb.org/3/movie/now_playing?api_key=${apikey}&language=en-US&page=${page}`;
+    //here we get an array of 20 movies
+    let fetchedData = await fetch(url);
+    let data = await fetchedData.json();
+    movieList = movieList.concat(data.results);
+
+    movieList.map(movie => {
+      movie.genres = movie.genre_ids.map(genre => genres.find(el => el.id === genre))
+    })
+    setMovies(movieList)
+  }
+  let loadMore = () => {
+    page++;
+    popularMovies()
+    setMovies(movieList);
+  };
+
+  const fetchGenres = async () => {
+    let secondUrl = `https://api.themoviedb.org/3/genre/movie/list?api_key=${apikey}&language=en-US`
+    let secondData = await fetch(secondUrl);
+    let secondResult = await secondData.json();
+    setGenres(secondResult.genres) // will change genres value
+  }
+  useEffect(() => {
+    fetchGenres();
+  }, []); // run once after mounted , mounted is different from re-rendered
+
+  useEffect(() => {
+    popularMovies();
+  }, [genres]);
+
+  //Input range slider
+
+  let searchByRate = (value) => {
+    let rounded = Math.round(value * 10) / 10
+
+    console.log("value", rounded)
+    setRate(value)
+
+    let filteredData = movieList.filter((movie) => movie.vote_average >= value);
+    setMovies(filteredData);
+  }
+
+  console.log(movies)
+  // Loading message
+  if (movies == null) {
```

Comment:
Someday you can use https://reactjs.org/docs/concurrent-mode-suspense.html but for now, this is fine.